explorer/ast/value.h

```diff
@@ -723,11 +739,17 @@ class FunctionType : public Value {
         impl_bindings_(std::move(impl_bindings)),
         is_initializing_(is_initializing) {}
 
-  struct ExceptSelf {};
-  FunctionType(ExceptSelf, const FunctionType& clone)
-      : FunctionType(std::nullopt, clone.parameters_, clone.generic_parameters_,
-                     clone.return_type_, clone.deduced_bindings_,
-                     clone.impl_bindings_, clone.is_initializing_) {}
+  struct ExceptSelf {
+    friend auto operator==(ExceptSelf, ExceptSelf) -> bool { return true; }
+    friend auto hash_value(ExceptSelf) -> llvm::hash_code {
+      return llvm::hash_combine(0);
+    }
+  };
```

Comment:
Optionally: make this type `Decompose`able
```suggestion
  struct ExceptSelf : public HashFromDecompose<ExceptSelf> {
    template <typename F> auto Decompose(F f) const { return f(); }
  };
```